sys/shell/shell.c

```diff
@@ -304,3 +304,17 @@ void shell_run(const shell_command_t *shell_commands, char *line_buf, int len)
         print_prompt();
     }
 }
+
+#ifdef MODULE_SHELL_COMMANDS
+/* Provide standard system() function */
+int system(const char *command)
```

Comment:
This as well.

Also I think this rather belongs into the POSIX module and `shell_commands` shouldn't be a dependency.

Reply:
This should have its own PR. I would not put it under POSIX, since system() is C99/C89 and placing it is the easiest way to make sure system() is tied to shell_commands.

Reply:
> since system() is C99/C89 and placing it is the easiest way to make sure system() is tied to shell_commands.

You can have shell commands without having the module `shell_commands` (which provides a number of *default* shell commands), so I don't understand the requirement for `shell_commands`.

Reply:
I you don't have shell_commands, then you don't have access to _any_ command from within system(), Because you have no way for your application to communicate a custom command list:

https://github.com/RIOT-OS/RIOT/blob/200c465f36ffee6b1958a59c80b8879d09ea1d2b/sys/shell/shell.c#L54-L62

Nothing will break, but no commands will be available.

In 2038, when we finally get XFAs, this won't be an issue (if we survive the UNIX timestamp overflow)

Reply:
> I you don't have shell_commands, then you don't have access to any command from within `system()`

Well, that seems to be more of an issue of the current implementation. So I see it as premature to establish a dependency that in reality should not be there (I would expect e.g. `system("udp …")` to work if I call it with `gnrc_networking`.

> In 2038, when we finally get XFAs, this won't be an issue (if we survive the UNIX timestamp overflow)

ztimer will handle the latter. ;-)